### ISSUE TITLE
Ability to set environment variables for run_* functions

### ISSUE THREAD

**Issue by nmaludy** [puppetlabs/bolt#1899]
## Use Case

As a bolt user i have a command/script that requires a password passed in on the command line. 

Example:
```shell
mongodump -u mongouser -p 'Abc123$' --gzip --archive=mongodump-$(date  --utc --iso-8601=seconds).gzip.archive
```

Ideally instead of embedding that password in the `command` or `arguments` string, i would like to set these as environment variables:

```shell
# set these secretly
export MONGO_PASSWORD='Abc123$'

# run the dump command
mongodump -u mongouser -p "${MONGO_PASSWORD}" --gzip --archive=mongodump-$(date  --utc --iso-8601=seconds).gzip.archive
```

Additionally, some commands that require passwords do not accept them on the CLI, for security reasons, and require the arguments be environment variables.

Example:
```shell
# these would be set secretly
export PGUSER='dbuser'
export PGPASSWORD='Abc123$'

# run the dump command
pg_dumpall --compress=9 -f pgdump-$(date  --utc --iso-8601=seconds).sql.gz
```

## Describe the Solution You Would Like
I would like the ability to set environment variables when calling the `run_command()` and `run_script()` functions (not sure if it makes sense for `run_task()`?). 

Example:
```puppet
run_command('mongodump -u mongouser -p "${MONGO_PASSWORD}" --gzip --archive=mongodump-$(date  --utc --iso-8601=seconds).gzip.archive',
            $targets,
            _environment => {
              MONGO_PASSWORD => $mongodb_password,
            })
```

Or the postgres example:
```puppet
run_command('pg_dumpall --compress=9 -f pgdump-$(date  --utc --iso-8601=seconds).sql.gz',
            $targets,
            _environment => {
              PGPASSWORD => $postgres_password,
              PGUSER => $postgres_username,
            })
```

### Additional Context

Maybe this deserves a different request, but also thinking about other "shell" things that people might want to set and `cwd` / `pwd` comes up ie. "what directory the command/script will be executed in". Example:

```
run_command('pg_dumpall --compress=9 -f pgdump-$(date  --utc --iso-8601=seconds).sql.gz',
            $targets,
            _env => {
              PGPASSWORD => $postgres_password,
              PGUSER => $postgres_username,
            },
            _cwd => '/mnt/backups')
```

**Comment by beechtom** [comment]
- We should not do this for `run_task`, as tasks have a clear API and there are ways to do this already
- We should do this for `run_command` and `run_script`
- This should be an `_env_vars` metaoption that accepts a hash
- We should error if a value is not a string

**Comment by beechtom** [comment]
Closed by #1943
